taxi/views.py

```diff
@@ -87,3 +108,15 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
     queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+
+
+@login_required
+def assign_car(request: HttpRequest, pk: int) -> HttpResponse:
```

Comment:
You can put this logic into `post` method of `CarDetailView`. The reason is that we don't need to create extra view and url endpoint to handle adding/deleting driver.
So the implementation will look something like this:

1. Update form instead on car detail template that will send value 'Add' or 'Delete' to post method to the cardetailview.
2. You override post method in this view and catch those values that you received in if condition and add/remove driver to car through related fields (basically you can copy logic from this method over there).
3. You will need to figure out how to get car and user in the view's post method to set related fields.